src/components/seeds/database/index.tsx

```diff
@@ -203,9 +204,10 @@ export default function Database(props: DatabaseProps): JSX.Element {
   const updateSearchColumns = useCallback(
     (columnNames?: string[]) => {
       if (columnNames) {
+        const columnInfo = columnsIndexed();
```

Comment:
This works just fine. 
Another option is to keep columns in the component state, the value won't change on every render. More of a hack.
Or [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) - which returns a memoized value (even though this is not really compute intensive).
Line 169 could be `const columns = useMemo(() => columnsIndexed(), []);`
I don't believe we have a particular preferred way of doing such things today - so anything goes.

Reply:
The reason I didn't just use the `columns` from the parent scope was that the linter complained that I was referencing `columns` in the `useEffect` callback but wasn't declaring it as a dependency. So I added it as a dependency, but then React seemed to get confused and go into an infinite recursive loop (which it detected and aborted). I guess maybe it uses reference equality, which would fail because `columnsIndexed()` returns a new array instance each time.

If I used `useMemo`, the reference equality problem would go away but I guess I'd have to pass the locale as a dependency to `useMemo` or it'd keep reusing the strings from the initial locale after you switched to a new one. That does seem a bit cleaner than what I'm doing here.

Reply:
That is correct, good call out - the locale would need to be a dependency.
I'm afraid the linter might complain that locale is not being directly referenced in the useMemo function. If that happens, the current version would be the best!!

Reply:
Tried `useMemo` and it has a similar lint problem where if I tell it the locale is a dependency, it complains that I have an unnecessary dependency because the actual function doesn't reference the locale. I could add a dummy reference to the locale to make the lint error go away but at that point I think it's no longer cleaner than the way I have it now.

Reply:
Ha, glad we reached the same conclusion!